packaging/core.re.template

```diff
@@ -701,7 +701,7 @@ acPostProcForDataCopyReceived(*leaf_resource) {}
 
 # =-=-=-=-=-=-=-
 # policy controlling when a dataObject is staged to cache from archive in a compound coordinating resource
-#  - the default is to stage when cache is not present ("when_necessary")
+#  - the default is to stage when cache is stale or not present ("when_necessary")
```

Comment:
@korydraughn / @trel - Are we okay changing the definition of `when_necessary` for 4.2? I know we had a discussion about this but can't remember where we landed. Just want to make sure this is acknowledged before this gets approved.

Reply:
I don't remember having a discussion about changing the behavior of this option.

@JustinKyleJames What lead to this change? Is it required for this PR?

Reply:
i believe we did decide that 'when_necessary' should kick in if EITHER the cache is missing (current, most likely behavior) OR the cache is stale (which we just hadn't tested/considered before - it's not clear how it would even happen in practice).

Nobody wants a stale cache, so let's clean up a weird edge case while we're here.

Reply:
I agree. I can't think of a downside to this.

Reply:
Okay, just wanted to make sure we were okay with the change for 4.2. I think it is an overall Good Thing.